integrations/unity-csharp/Assets/VDK/UDUtilities.cs

```diff
@@ -46,21 +63,90 @@ public static vdkRenderInstance[] getUDSInstances()
             vdkRenderInstance[] modelArray = new vdkRenderInstance[objects.Length];
             for (int i = 0; i < objects.Length; ++i)
             {
-                Component component = objects[i].GetComponent("UDSModel");
-                UDSModel model = component as UDSModel;
+                UDSModel model = (UDSModel) objects[i].GetComponent("UDSModel");
 
                 if (!model.isLoaded)
                     model.LoadModel();
 
                 if (model.isLoaded)
                 {
                     modelArray[count].pointCloud = model.udModel.pModel;
-                    modelArray[count].worldMatrix = UDUtilities.GetUDMatrix(model.pivotTranslation * model.modelScale * objects[i].transform.localToWorldMatrix * model.pivotTranslation.inverse);
+                    Transform localTransform = objects[i].transform;
+
+                    modelArray[count].worldMatrix = UDUtilities.GetUDMatrix(
+                            Matrix4x4.TRS(model.transform.position, model.transform.rotation, model.transform.localScale) *
+                            model.modelToPivot
+                        );
```

Comment:
Indentation and strangely multiline?

Reply:
Not a fan of cramming complex expressions into one line seeing as whitespace doesn't cost anything... Are we not allowed multiline expressions?